lib/index.js

```diff
@@ -437,7 +437,7 @@ module.exports.renderSync = function(opts) {
     return result;
   }
 
-  throw util._extend(new Error(), JSON.parse(result.error));
+  throw assign(new Error(), JSON.parse(result.error));
```

Comment:
I'm not following what the difference is between this and throwing an error
as I posted earlier?

On 8 Sep 2016 9:24 PM, "Adeel Mujahid" <notifications@github.com> wrote:

> In lib/index.js
> <https://github.com/sass/node-sass/pull/1711#discussion_r77988340>:
>
> > @@ -437,7 +437,7 @@ module.exports.renderSync = function(opts) {
> >      return result;
> >    }
> >
> > -  throw util._extend(new Error(), JSON.parse(result.error));
> > +  throw assign(new Error(), JSON.parse(result.error));
>
> Yup, the idea was to have the error object thrown (or return in case of
> asynchornous call) to the caller. We tried to avoid JSON parsing on native
> side and casting it to object in JavaScript using util._extend (without
> having to know what is in the object, line/column/message).
>
> lodash.assign approach is the best backward compatible solution, if we
> are already depending on that lib.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/sass/node-sass/pull/1711/files/61f42dee63a8175cf2ccaa35e41ee243b3875074#r77988340>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAjZWCcVzPeaZpxPQDIx4h0mJ49uRhHhks5qn_B2gaJpZM4J2hC7>
> .
>